### PR TITLE
MCH: moved time calibration to digits filter

### PR DESCRIPTION
**Pull request by aferrero2707**
The time calibration is controlled by the `MCHDigitFilterParam.timeOffset` configuration parameter, that specifies the global offset (in BC units) applied to the time stamp of digits and ROFs.

### CODE REVIEW

**Comment by aphecetche** [comment]
@aferrero2707, if you agree, I would propose that : 

1 - we merge https://github.com/AliceO2Group/AliceO2/pull/8737 to get the `DigitFilterParam` in place
2 - we introduce the digit noise filtering (PR to be opened after 1 is merged, see https://github.com/AliceO2Group/AliceO2/commit/65b0fb905883fbc501cab7e2d763996030145149) 
3 - we rebase this one on (1 or 2) to benefit from the DigitFilterParam

or equivalently, 1 - 3 -2 (2 and 3 being different filtering features)

what do you think ?

**Comment by aphecetche** [comment]
@aferrero2707, if you agree, I would propose that : 

1 - we merge https://github.com/AliceO2Group/AliceO2/pull/8737 to get the `DigitFilterParam` in place
2 - we introduce the digit noise filtering (PR to be opened after 1 is merged, see https://github.com/AliceO2Group/AliceO2/commit/65b0fb905883fbc501cab7e2d763996030145149) 
3 - we rebase this one on (1 or 2) to benefit from the DigitFilterParam

or equivalently, 1 - 3 -2 (2 and 3 being different filtering features)

what do you think ?

**Comment by aferrero2707** [comment]
I think 1 and 2 are more urgent than this PR, so we can rebase this one once the other two are merged.

**Comment by shahor02** [comment]
> I think 1 and 2 are more urgent than this PR, so we can rebase this one once the other two are merged.

@aferrero2707 OK, would you check/validate https://github.com/AliceO2Group/AliceO2/pull/8737 ? I'll merge it once fullCI is passed

**Comment by aferrero2707** [comment]
@aphecetche all checks are green, could you please have a look if the modifications are OK for you? meanwhile I will commit a small update of the documentation.
We should probably also consider a unit test dedicated to the digits filter, but I would propose to make a dedicated PR for that.

File: Detectors/MUON/MCH/DigitFiltering/src/DigitFilteringSpec.cxx
Comment (pillot) [inline Detectors/MUON/MCH/DigitFiltering/src/DigitFilteringSpec.cxx]: this data member is not used

**Review by pillot — COMMENTED** [review]
Hi @aferrero2707 ,
Appart from the little change below, shouldn't the configurable parameter CoDecParam::Instance().sampaBcOffset be removed?
In which case it also has to be removed from the few places where it is still used (e.g. testSampaBunchCrossingCounter.cxx, codec-closure-test.sh and testClosureCoDecDigit.cxx).

**Comment by aferrero2707** [comment]
@pillot you are absolutely right! It seems I stopped in the middle of the code cleanup to remove the `sampaBcOffset`.

File: Detectors/MUON/MCH/DigitFiltering/src/DigitFilteringSpec.cxx
Comment (aferrero2707) [reply Detectors/MUON/MCH/DigitFiltering/src/DigitFilteringSpec.cxx]: Removed.

**Review by aphecetche — APPROVED** [review]

**Pull request merged**
